src/main/java/bruno/TaskList.java

```diff
@@ -0,0 +1,255 @@
+        }
+        tasks.add(new ToDo(task.substring(task.indexOf(" ") + 1)));
+        String taskInfo = "Woof. I have added this task:\n" + tasks.get(tasks.size() - 1).getString();
+        storage.writeToFile(this);
+        return taskInfo;
+    }
+
+    /**
+     * Adds a task of type Deadline to the tasks of tasks.
+     *
+     * @param task The user input for the task.
+     * @throws BrunoException Thrown if user does not provide description or deadline date/time.
+     */
+    public String addDeadline(String task) throws BrunoException {
+        if (task.split(" ").length == 1 || task.indexOf('/') == 9) {
+            throw new BrunoEmptyException(task.split(" ")[0]);
+        }
+        if (!task.substring(task.indexOf("deadline") + 1).contains("/by")) {
+            throw new BrunoMissingDeadlineException();
+        }
+        tasks.add(new Deadline(task.substring(task.indexOf(' ') + 1, task.indexOf('/') - 1),
+                task.substring(task.lastIndexOf('/') + 4)));
+        storage.writeToFile(this);
+        String taskInfo = "Woof. I have added this task:\n" + tasks.get(tasks.size() - 1).getString();
+        return taskInfo;
+    }
+
+    /**
+     * Adds a task of type Event to the tasks of tasks.
+     *
+     * @param task The user input for the task.
+     * @throws BrunoException Thrown if the user does not provide description or start/end time.
+     */
+    public String addEvent(String task) throws BrunoException {
+        if (task.split(" ").length == 1 || task.indexOf('/') == 6) {
+            throw new BrunoEmptyException(task.split(" ")[0]);
+        }
+        if (!task.substring(task.indexOf("event") + 1).contains("/from") || !task.substring(
+                task.indexOf("event") + 1).contains("/to")) {
+            throw new BrunoMissingEventException();
+        }
+        tasks.add(new Event(task.substring(task.indexOf(' ') + 1, task.indexOf('/') - 1),
+                task.substring(task.indexOf("from") + 5, task.lastIndexOf('/') - 1),
+                task.substring(task.indexOf("to") + 3)));
+        storage.writeToFile(this);
+        String taskInfo = "Woof. I have added this task:\n" + tasks.get(tasks.size() - 1).getString();
+        return taskInfo;
+    }
+
+    /**
+     * Marks a task as done from the tasks of tasks.
+     *
+     * @param task The user input for the task.
+     * @throws BrunoException Thrown if the user tries to mark an invalid task.
+     */
+    public String markTask(String task) throws BrunoException {
+        String markVal = task.split(" ")[1];
+        try {
+            int a = Integer.parseInt(markVal);
+        } catch (NumberFormatException e) {
+            throw new BrunoIntegerMismatchException("mark");
+        }
+        if (Integer.parseInt(markVal) > tasks.size()) {
+            throw new BrunoIndexOutOfBoundsException("mark");
+        }
+        if (Integer.parseInt(markVal) < 0) {
+            throw new BrunoNegativeArgException("mark");
```

Comment:
I like how you have 2 different exceptions to be thrown for either out of bound or negative index 👍